docs/serial_driver.md

```diff
@@ -68,3 +70,167 @@ You must also enable the ChibiOS `SERIAL` feature:
+#define SELECT_SOFT_SERIAL_SPEED 1 // or 0, 2, 3, 4, 5
+                                   //  0: about 460800 baud
+                                   //  1: about 230400 baud (default)
+                                   //  2: about 115200 baud
+                                   //  3: about 57600 baud
+                                   //  4: about 38400 baud
+                                   //  5: about 19200 baud
+#define SERIAL_USART_DRIVER UARTD1 // USART driver of TX and RX pin. default: UARTD1
+#define SERIAL_USART_TX_PAL_MODE 7 // Pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 7
+#define SERIAL_USART_RX_PAL_MODE 7 // Pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 7
+#define SERIAL_USART_TIMEOUT 100 // USART driver timeout. default 100
+```
+
+You must also enable the ChibiOS `UART` with blocking api feature:
+* In your board's halconf.h: `#define HAL_USE_USAR TRUE` and `#define UART_USE_WAIT TRUE`
+* In your board's mcuconf.h: `#define STM32_UART_USE_USARTn TRUE` (where 'n' matches the peripheral number of your selected USART on the MCU)
+
+Do note that the configuration required is for the `UART` peripheral, not the `SERIAL` peripheral.
+
+#### Pins for USART Peripherals with Alternate Functions for selected STM32 MCUs
+
+##### STM32F072 [Datasheet](https://www.st.com/resource/en/datasheet/stm32f072c8.pdf)
+
+Pin Swap available: :heavy_check_mark:
+
+USART1:
+| Pin  | Function | Mode |
+| ---- | -------- | ---- |
+| PA9  | TX       | AF1  |
+| PA10 | RX       | AF1  |
+| PB6  | TX       | AF0  |
+| PB7  | RX       | AF0  |
+
+USART2:
+| Pin  | Function | Mode |
+| ---- | -------- | ---- |
+| PA2  | TX       | AF1  |
+| PA3  | RX       | AF1  |
+| PA14 | TX       | AF1  |
+| PA15 | RX       | AF1  |
+
+USART3:
+| Pin  | Function | Mode |
+| ---- | -------- | ---- |
+| PB10 | TX       | AF4  |
+| PB11 | RX       | AF4  |
+| PC4  | TX       | AF1  |
+| PC5  | RX       | AF1  |
+| PC10 | TX       | AF1  |
+| PC11 | RX       | AF1  |
+| PD8  | TX       | AF0  |
+| PD9  | RX       | AF0  |
+
+USART4:
+| Pin | Function | Mode |
+| --- | -------- | ---- |
+| PA0 | TX       | AF4  |
+| PA1 | RX       | AF4  |
+
+##### STM32F103 Medium Density (C8-CB) [Datasheet](https://www.st.com/resource/en/datasheet/stm32f103c8.pdf)
+
+Pin Swap available: N/A
+
+TX Pin is always Alternate Function Push-Pull, RX Pin is always regular input pin for any USART peripheral. **For STM32F103 no additional Alternate Function configuration is necessary. QMK is already configured.**
+
+Pin remapping:
+
+The pins of USART Peripherals use default Pins that can be remapped to use other pins using the AFIO registers. Default pins are marked **bold**. Here is an example how you can remap the pins for USART1.
+
+```c
+// Remap USART1 from pins PA9/PA10 to PB6/PB7
+void keyboard_pre_init_user(void) {
```

Comment:
Would rather this be a setting, via a define, than this.

Reply:
Add defines for USART remapping. This affects half-duplex as well as full-duplex.